### PR TITLE
KOGITO-1704 + KOGITO-1124 - Serverless Workflow - Support for Relay and Switch States

### PR DESCRIPTION
**Pull request by tsurdilo**
* Add support for Relay state
* Add support for Switch state
* Update support for script function
* Add missing tests

### CODE REVIEW

File: jbpm/jbpm-serverless-workflow/pom.xml [hunk 1]
```diff
@@ -61,6 +61,14 @@
       <groupId>com.fasterxml.jackson.dataformat</groupId>
       <artifactId>jackson-dataformat-yaml</artifactId>
     </dependency>
+    <dependency>
+      <groupId>com.jayway.jsonpath</groupId>
+      <artifactId>json-path</artifactId>
+    </dependency>
+    <dependency>
+      <groupId>com.jayway.jsonpath</groupId>
+      <artifactId>json-path-assert</artifactId>
```
Comment (cristianonicolai) [inline jbpm/jbpm-serverless-workflow/pom.xml:70]: test scope?

Comment (tsurdilo) [reply jbpm/jbpm-serverless-workflow/pom.xml]: updated. thanks!

**Comment by tsurdilo** [comment]
@cristianonicolai @ricardozanini I am really sorry but I added to this PR also support for switch state (xor gateways). It was just too hard to create over weekend a separate pr which depends on this one without this one being merged already. If you don't mind please review again. Again, I am sorry for the extra work.

**Comment by tsurdilo** [comment]
@evacchi Hi, can you please review? Need this to go in to move on with further enhancements. Thanks!

File: jbpm/jbpm-serverless-workflow/src/main/java/org/jbpm/serverless/workflow/parser/core/ServerlessWorkflowFactory.java [hunk 2]
```diff
@@ -55,7 +56,8 @@
     private static final String DEFAULT_VAR = "Var";
     private static final String JSON_NODE = "com.fasterxml.jackson.databind.JsonNode";
     private static final String DEFAULT_WORKFLOW_VAR = "workflowdata";
-    private static final Set<String> DEFAULT_IMPORTS = new HashSet<String>(Arrays.asList(JSON_NODE));
+    private static final Set<String> DEFAULT_IMPORTS = new HashSet<>(Arrays.asList("org.jbpm.serverless.workflow.parser.util.ServerlessWorkflowUtils"));
```
Comment (evacchi) [inline jbpm/jbpm-serverless-workflow/src/main/java/org/jbpm/serverless/workflow/parser/core/ServerlessWorkflowFactory.java:59]: any reason for this to be a Set specifically? You may use a collection and avoid the `new HashSet()` altogether.

Comment (tsurdilo) [reply jbpm/jbpm-serverless-workflow/src/main/java/org/jbpm/serverless/workflow/parser/core/ServerlessWorkflowFactory.java]: Oops - nice catch!!! The default imports were not needed. Removed. Thanks!

**Comment by tsurdilo** [comment]
Jenkins fail is on Kogito :: Management Console, not related to these changes

**Comment by ricardozanini** [comment]
Are we ready? @evacchi

**Pull request merged**